File: C#/CocktailShakerSort.cs
using System;

class CocktailShakerSort
{
    public static void Sort(int[] arr)
    {
        int n = arr.Length;
        bool swapped;
        do
        {
            swapped = false;
            for (int i = 0; i < n - 1; i++)
            {
                if (arr[i] > arr[i + 1])
                {
                    int temp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = temp;
                    swapped = true;
                }
            }
            if (!swapped) break;

            swapped = false;
            for (int i = n - 2; i >= 0; i--)
            {
                if (arr[i] > arr[i + 1])
                {
                    int temp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = temp;
                    swapped = true;
                }
            }
        } while (swapped);
    }
    
    public static void Main()
    {
        int[] arr = { 5, 3, 8, 4, 2, 7, 1, 6 };
        Sort(arr);
        Console.WriteLine(string.Join(" ", arr));
    }
}

File: PHP/cocktail_shaker_sort.php
<?php
function cocktailShakerSort(&$arr) {
    $n = count($arr);
    $swapped = true;
    $start = 0;
    $end = $n - 1;

    while ($swapped) {
        $swapped = false;

        for ($i = $start; $i < $end; $i++) {
            if ($arr[$i] > $arr[$i + 1]) {
                $temp = $arr[$i];
                $arr[$i] = $arr[$i + 1];
                $arr[$i + 1] = $temp;
                $swapped = true;
            }
        }

        if (!$swapped) {
            break;
        }

        $swapped = false;
        $end--;

        for ($i = $end - 1; $i >= $start; $i--) {
            if ($arr[$i] > $arr[$i + 1]) {
                $temp = $arr[$i];
                $arr[$i] = $arr[$i + 1];
                $arr[$i + 1] = $temp;
                $swapped = true;
            }
        }

        $start++;
    }
}

$array = [5, 3, 8, 4, 2, 7, 1, 10, -8, 1, -11];

cocktailShakerSort($array);

echo "Sorted Array:\n";
echo "[" . implode(", ", $array) . "]\n";
